### PR TITLE
CStor autogenerated CRD code

### PR DESCRIPTION
**Pull request by gkGaneshR**
1. Why is this change necessary ?
fixes: openebs/openebs#1363, openebs/openebs#1339

2. How does this change address the issue ?
pkg/apis...will have the struct type definition and pkg/client
should have clientset, informer and lister autogenerated.

Signed-off-by: gkGaneshR <gkganesh126@gmail.com>

**What this PR does / why we need it**:
Autogenerated crd code for crd watcher.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
openebs/openebs#1363, openebs/openebs#1339

**Special notes for your reviewer**:
Need to verify types.go file, register.go and check if namespace parameter is removed in any CStorPool/CStorVolumeReplica functions autogenerated.
a). Link to yaml files: openebs/openebs@69ab188
b).Under openebs/maya, Run : buildscripts/update-codegen.sh will create autogenerated code for crds.
```
ganesh@ganesh-ThinkPad-E470:~/gocode/src/github.com/openebs/maya$ buildscripts/update-codegen.sh 
Generating deepcopy funcs
Generating clientset for openebs.io:v1alpha1 at github.com/openebs/maya/pkg/client/clientset
Generating listers for openebs.io:v1alpha1 at github.com/openebs/maya/pkg/client/listers
Generating informers for openebs.io:v1alpha1 at github.com/openebs/maya/pkg/client/informers
```
c). Input to autogen is kept under pkg/apis -> types.go specifies type struct definition, register.go needs to be registered with the variable names and rest of the files are generated.
Output is generated at pkg/client/clientset, pkg/client/informers/ pkg/client/listers

d).pkg/signals will be used in business logic for syncing stop channel. It comes as package along with autogen component, So it was decided to keep under pkg.

### CODE REVIEW

**Comment by gkGaneshR** [comment]
This commit has added certain fields in storagepoolclaim and storagepool also, since it was decided to keep all crds in same file and git stash can't separate them. We can remove them, while cleaning unused code like cmd/maya-agent, gnumakefile, etc.

File: pkg/apis/openebs.io/v1alpha1/types.go [hunk 1]
```diff
@@ -13,14 +13,17 @@ import (
 // StoragePoolClaim describes a StoragePoolClaim.
 type StoragePoolClaim struct {
 	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
+	metav1.ObjectMeta `json:"metadata",omitempty`
```
Comment (kmova) [inline pkg/apis/openebs.io/v1alpha1/types.go]: what was the issue with the above line?

File: pkg/apis/openebs.io/v1alpha1/types.go [hunk 2]
```diff
@@ -142,3 +150,77 @@ type Task struct {
...
+// CStorVolumeReplica describes a cstor pool resource created as custom resource
+type CStorVolumeReplica struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata",omitempty`
+	Spec              CStorVolumeReplicaSpec `json:"spec"`
+}
+
+// CStorVolumeReplicaSpec is the spec for a CStorVolumeReplica resource
+type CStorVolumeReplicaSpec struct {
+	CstorControllerIP string `json:"cstorControllerIP"`
```
Comment (kmova) [inline pkg/apis/openebs.io/v1alpha1/types.go]: should this be CStor?

File: pkg/client/clientset/versioned/typed/openebs/v1alpha1/generated_expansion.go [hunk 3]
```diff
@@ -15,6 +15,10 @@ limitations under the License.
 */
 package v1alpha1
 
+type CStorPoolExpansion interface{}
```
Comment (kmova) [inline pkg/client/clientset/versioned/typed/openebs/v1alpha1/generated_expansion.go]: What are these object with suffix Expansion?

File: pkg/apis/openebs.io/v1alpha1/types.go (hunk 1 above)
Comment (gkGaneshR) [reply pkg/apis/openebs.io/v1alpha1/types.go]: That was just for learning-trial and error. will revert

File: pkg/client/clientset/versioned/typed/openebs/v1alpha1/generated_expansion.go (hunk 3 above)
Comment (gkGaneshR) [reply pkg/client/clientset/versioned/typed/openebs/v1alpha1/generated_expansion.go]: It is generated by autogen. It reads the type struct name and does some string processing and create new variables.

File: pkg/apis/openebs.io/v1alpha1/types.go (hunk 2 above)
Comment (gkGaneshR) [reply pkg/apis/openebs.io/v1alpha1/types.go]: will change

**Comment by kmova** [comment]
@gkGaneshR I think overall the changes are ok. For better understanding of the PR, can you include the following details in the description under the notes for reviewer:

(a) What are the YAMLs used for auto-generation code. Either provide a link to that YAML files or paste the content. 
(b) What is the command used to auto-generate. If the command and output are pasted, it will help to know exactly what files are auto generated. 
(c) In the next PRs, it would be good if the auto-generated files are checked in a separate commit. For this commit, specify the list of files that were auto-generated and files that were modified.

File: pkg/apis/openebs.io/v1alpha1/types.go [hunk 4]
```diff
@@ -142,3 +150,77 @@ type Task struct {
...
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=cstorvolumereplica
+// +genclient:nonNamespaced
+
+// CStorVolumeReplica describes a cstor pool resource created as custom resource
+type CStorVolumeReplica struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata",omitempty`
```
Comment (prateekpandey14) [inline pkg/apis/openebs.io/v1alpha1/types.go]: Double quotes should come at the end for whole json tag like `json:"metadata ,omitempty"`

File: pkg/apis/openebs.io/v1alpha1/types.go [hunk 5]
```diff
@@ -142,3 +150,77 @@ type Task struct {
...
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +genclient:nonNamespaced
+// +resource:path=cstorpool
+
+// CStorPool describes a cstor pool resource created as custom resource.
+type CStorPool struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata",omitempty`
```
Comment (prateekpandey14) [inline pkg/apis/openebs.io/v1alpha1/types.go]: Double quotes should come at the end for whole json tag like `json:"metadata,omitempty"`

File: pkg/apis/openebs.io/v1alpha1/types.go (hunk 4 above)
Comment (gkGaneshR) [reply pkg/apis/openebs.io/v1alpha1/types.go]: will change

File: pkg/apis/openebs.io/v1alpha1/types.go (hunk 5 above)
Comment (gkGaneshR) [reply pkg/apis/openebs.io/v1alpha1/types.go]: will change

**Comment by gkGaneshR** [comment]
a). Link to yaml files: https://github.com/openebs/openebs/pull/1341/commits/69ab188146cc9d01af2e762f0e6ac04e27add3c1
b).Under openebs/maya, Run : buildscripts/update-codegen.sh will create autogenerated code for crds.
c). Input to autogen is kept under pkg/apis -> types.go specifies type struct definition, register.go needs to be registered with the variable names and rest of the files are generated.
Output is generated at pkg/client/clientset, pkg/client/informers/ pkg/client/listers

**Comment by kmova** [comment]
@gkGaneshR Looks good. one last comment. the pkg/signals? Why is it being added and what should be the copy right header for these files?

**Comment by gkGaneshR** [comment]
pkg/signals will be used in business logic for syncing stop channel. It comes as package along with autogen component, So it was decided to keep under pkg.

**Pull request merged**